src/ORM/Table.php

```diff
@@ -1436,6 +1435,36 @@ public function get($primaryKey, $options = [])
     }
 
     /**
+     * Handles the logic executing of a worker inside a transaction.
+     *
+     * @param bool $atomic Whether to execute the worker inside a database transaction.
+     * @param callable $worker The worker that will run inside the transaction.
+     * @return mixed
+     */
+    protected function _executeTransaction($atomic, callable $worker)
+    {
+        if ($atomic) {
+            return $this->connection()->transactional(function () use ($worker) {
+                return call_user_func($worker);
+            });
+        }
+
+        return call_user_func($worker);
+    }
+
+    /**
+     * Checks if the caller would have executed a commit on a transaction.
+     *
+     * @param bool $atomic True if an atomic transaction was used.
+     * @param bool $primary True if a primary was used.
+     * @return bool Returns true if a transaction was committed.
+     */
+    protected function _getCommitUsed($atomic, $primary)
```

Comment:
@lorenzo hard to name the method, sorry. TRUE if the "previous" operation issued a `COMMIT;` for a transaction. It's a logic function.